### ISSUE TITLE
Error fetching trade

### ISSUE THREAD

**Issue by HarlamovEvgeniy** [Uniswap/widgets#627]
**Bug Description**
The problem occurs on the widget screen. The endless error "Error fetching trade"

![image](https://github.com/Uniswap/widgets/assets/104944098/c893d522-8a83-4a52-9958-7bbbbfd15325)

Versions:
Vite: 4.5.1
React: 18.2.0
TypeScript: 5.3.3
@uniswap/widgets: 2.59.0
...

Expected Behavior

**Comment by HarlamovEvgeniy** [comment]
My vite.config.ts

`import react from '@vitejs/plugin-react'
import { fileURLToPath } from 'url'
import { defineConfig } from 'vite'
import { createHtmlPlugin } from 'vite-plugin-html'
import svgLoader from 'vite-svg-loader'
import { chunkSplitPlugin } from 'vite-plugin-chunk-split';
import { nodePolyfills } from "vite-plugin-node-polyfills"
import path from "path";

export default defineConfig({
  plugins: [
    react(),
    createHtmlPlugin({
      minify: true,
      entry: '/src/main.tsx',
      template: '/index.html',
    }),
    svgLoader(),
    chunkSplitPlugin({
      strategy: 'unbundle',
    }),
    nodePolyfills()
  ],
  server: {
    port: 3000,
    open: true,
  },
  optimizeDeps: {
    esbuildOptions: {
      define: {
        global: 'globalThis'
      }
    }
  },
  resolve: {
    alias: {
      "@": fileURLToPath(new URL('./src', import.meta.url)),
      jsbi: path.resolve(__dirname, "./node_modules/jsbi/dist/jsbi-cjs.js"),
      process: "process/browser",
      buffer: "buffer",
      crypto: "crypto-browserify",
      stream: "stream-browserify",
      assert: "assert",
      http: "stream-http",
      https: "https-browserify",
      os: "os-browserify",
      url: "url",
      util: "util",
    }
  },
  build: {
    outDir: path.join(__dirname, "build"),
    sourcemap: false,
    minify: true,
    commonjsOptions: {
      transformMixedEsModules: true,
    }
  },
})`

**Comment by Asharp19** [comment]
Same Guys Been facing this. Got to launch a frontend for my token was thinking could add a widget before liquidity pool is set since short on time. 
Please fix it.

**Comment by Antonio100898** [comment]
Same

**Comment by SerovMihail** [comment]
We are facing the same issue with the same configuration. Any solution?

**Comment by HarlamovEvgeniy** [comment]
The only solution that helped me was to implement my widget using the SDK-v3

**Comment by rishabh-threejs** [comment]
> The only solution that helped me was to implement my widget using the SDK-v3

IKR!. Would have been way more convenient with the widget though.

**Comment by Antonio100898** [comment]
May I ask you to share the code with me? I would appreciate a lot if you
could just give a direction of what I need to change for my own project
(provider/rpc addresses etc.)

среда, 31 января 2024 г. пользователь HarlamovEvgeniy <
***@***.***> написал:

**Comment by Chi111** [comment]
same

**Comment by HarlamovEvgeniy** [comment]
Any solution?

**Comment by Chi111** [comment]
Same!
"react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.8.0",
    "@uniswap/widgets": "2.59.0",
    "@metamask/detect-provider": "2.0.0",
    "ethers": "5.7.2",

**Comment by Chi111** [comment]
> The only solution that helped me was to implement my widget using the SDK-v3

Can You Give Me a demo? My project is still wrong

**Comment by HarlamovEvgeniy** [comment]
> Can You Give Me a demo? My project is still wrong

I can't share it because the reason is in the smart-order-router library, and I couldn't make my own implementation:(

**Comment by dziobakwszafie** [comment]
Any update for that? I;m facing the same problem with Next.js

**Comment by bart-jablonski** [comment]
My summary of the issue so far is as follows:

This issue occurs at least on the following browsers:
- Chrome 121.0.6167.139 (Official Build) (64-bit)
- Firefox 122.0 (64-bit)

This issue occurs in all the latest bundlers:
 - Vite (Rollup)
 - CRA (Webpack5)
 - NextJS (Webpack5)
 
The core issue is in this package: https://github.com/foliojs/brotli.js which is a dependency of `@uniswap/smart-order-router` package. There is an old issue raised for that so it's unclear why this resurfaced here just now:
https://github.com/foliojs/brotli.js/issues/20

The workaround we found so far (dirty one) is to monkey patch the window object and tell it that the Browser.T function exists:

```ts
window.Browser = {
    T: () => {}
}
```

**Comment by Wunsz** [comment]
My summary of the issue so far is as follows:

This issue occurs at least on the following browsers:
- Chrome 121.0.6167.139 (Official Build) (64-bit)
- Firefox 122.0 (64-bit)

This issue occurs in all the latest bundlers:
 - Vite (Rollup)
 - CRA (Webpack5)
 - NextJS (Webpack5)
 
The core issue is in this package: https://github.com/foliojs/brotli.js which is a dependency of `@uniswap/smart-order-router` package. There is an old issue raised for that so it's unclear why this resurfaced here just now:
https://github.com/foliojs/brotli.js/issues/20

The workaround we found so far (dirty one) is to monkey patch the window object and tell it that the Browser.T function exists:

```ts
window.Browser = {
    T: () => {}
}
```

**Comment by fabdarice** [comment]
Following

**Comment by smartbart1** [comment]
same

**Comment by LilNait-S** [comment]
> Any update for that? I;m facing the same problem with Next.js

The same issue but with @uniswap/smart-order-router

**Comment by lisabeyy** [comment]
Same issue and bunches of error when installing the widget, this definitely not working with nextjs 14 and the given workaround

**Comment by russell0430** [comment]
same issue with nextjs@14

**Comment by Dariocoding** [comment]
same issue with nextjs@14 x2

**Comment by Dariocoding** [comment]
¿What should i do? Shit

**Comment by Wunsz** [comment]
What works for us is:

1. Specify `"use client";` at the top of the file where  you use the SwapWidget component
2. Before the component is defined, we add:
```
if (typeof window !== "undefined") {
// @ts-ignore
  window.Browser = {
    T: () => {
    }
  };
}
```
3. Add this to your next.js config:
```
/** @type {import('next').NextConfig} */
// 👇️ assumes you use Webpack 5
module.exports = {
  webpack: (config) => {
    config.resolve.fallback = {
      fs: false,
      path: false,
      Browser: false,
    };

    return config;
  },
  externals: ["pino-pretty"],
};
```

Sample component code:
```tsx
"use client";

import React from "react";
import { SwapWidget } from "@uniswap/widgets";


if (typeof window !== "undefined") {
    // @ts-ignore
      window.Browser = {
        T: () => {
        }
      };
    }
    

const SwapComponent = () => <SwapWidget />;

export default SwapComponent;
```

And that actually made it work some time ago but you will now hit:
![image](https://github.com/Uniswap/widgets/assets/11278533/9f6281d2-35f3-42ee-981b-b8e190c78802)


For now you can try my fork here:
https://www.npmjs.com/package/@wunsz-4soft/uniswap-widgets

But I do not guarantee anything, especially its maintenance.

**Comment by linxux** [comment]
Same issue on the version - `2.59.0`.
And the @Wunsz 's workaround is not working for me.

```
    "@uniswap/widgets": "^2.59.0",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "react-redux": "^9.1.2",
    "react-scripts": "5.0.1",
    "typescript": "^4.4.2",
    "web-vitals": "^2.1.0"
```

**Comment by limian1858** [comment]
same

**Comment by GolfredoPerezFernandez** [comment]
what should we do!

**Comment by hieppd-slz** [comment]
any solution for nextjs14 guys?

**Comment by dziobakwszafie** [comment]
> any solution for nextjs14 guys?

nope, that library is a big piece of you know what, recommend to do it on your own, will be still faster then fixing all bugs in there
